peer/channel.go

```diff
@@ -308,7 +306,13 @@ func (c *Channel) isClosed() bool {
 func (c *Channel) waitOpened() error {
 	select {
 	case <-c.opened:
-		return nil
+		// Re-check to prioritize the closed channel.
+		select {
+		case <-c.closed:
+			return c.closeError
+		default:
+			return nil
+		}
```

Comment:
:+1: An alternative. Exact same code, but reuses that `isClosed` function
```suggestion
		if c.isClosed() {
			return c.closeError
		}
		return nil
```

Reply:
Nice! 😎